geode-core/src/main/java/org/apache/geode/cache/query/security/JavaBeanAccessorMethodAuthorizer.java

```diff
@@ -0,0 +1,173 @@
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(Cache cache, Set<String> allowedPackages) {
+    Objects.requireNonNull(cache, NULL_CACHE_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = new RestrictedMethodAuthorizer(cache);
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Creates a {@code JavaBeanAccessorMethodAuthorizer} object and initializes it so it can be
+   * safely used in a multi-threaded environment.
+   * <p/>
+   *
+   * @param restrictedMethodAuthorizer the default {@code RestrictedMethodAuthorizer} to use.
+   * @param allowedPackages the packages containing classes for which 'is' and 'get' methods will
+   *        be authorized.
+   */
+  public JavaBeanAccessorMethodAuthorizer(RestrictedMethodAuthorizer restrictedMethodAuthorizer,
+      Set<String> allowedPackages) {
+    Objects.requireNonNull(restrictedMethodAuthorizer, NULL_AUTHORIZER_MESSAGE);
+    Objects.requireNonNull(allowedPackages, NULL_PACKAGE_MESSAGE);
+    this.restrictedMethodAuthorizer = restrictedMethodAuthorizer;
+    this.allowedPackages = Collections.unmodifiableSet(allowedPackages);
+  }
+
+  /**
+   * Executes the authorization logic to determine whether the {@code method} is allowed to be
+   * executed on the {@code target} object instance.
+   * <p/>
+   *
+   * @param method the {@link Method} that should be authorized.
+   * @param target the {@link Object} on which the {@link Method} will be executed.
+   * @return {@code true} if the {@code method} can be executed on on the {@code target} instance,
+   *         {@code false} otherwise.
+   *
+   * @see org.apache.geode.cache.query.security.MethodInvocationAuthorizer
+   */
+  @Override
+  public boolean authorize(Method method, Object target) {
+
+    // First ensure that the given method exists for the given target
+    if (!Arrays.asList(target.getClass().getMethods()).contains(method)) {
+      return false;
+    }
+
+    // Return false for known dangerous methods.
+    if (restrictedMethodAuthorizer.isKnownDangerousMethod(method, target)) {
+      return false;
+    }
+
+    String packageName = target.getClass().getPackage().getName();
+
+    // If the target object belongs to the 'org.apache.geode' package, delegate to the default
+    // authorizer.
+    if (packageName.startsWith(GEODE_BASE_PACKAGE)) {
+      return restrictedMethodAuthorizer.isAllowedGeodeMethod(method, target);
+    }
+
+    String methodName = method.getName();
+
+    // Return true if the the object belongs to an allowed package and the method starts with
+    // exactly 'get' or 'is' followed by a non-lowercase character (to prevent matching with
+    // methods that might have names like 'getawayDate' or 'islandName' etc.)
+    if (Pattern.matches("^(get|is)($|[^A-Z])+", methodName)) {
```

Comment:
Great catch regarding non accessor methods starting with `is|get`!!.
One minor change, though, `Pattern.matches` internally compiles the regular expression every single time, resulting in a waste of processing time while evaluating the OQL:

```
public static boolean matches(String regex, CharSequence input) {
  Pattern p = Pattern.compile(regex);
  Matcher m = p.matcher(input);
  return m.matches();
}
```

Instead of doing that, I'd suggest to store and reuse the compiled pattern within a `private final static` field, something like below:
```
Pattern pattern = Pattern.compile("^(get|is)($|[^A-Z])+");
pattern.matcher(methodName).matches();
```